packages/std/src/metadata.rs

```diff
@@ -21,5 +22,296 @@ pub struct DenomMetadata {
+        });
+
+        let metadata_with_null_denom_units: DenomMetadata =
+            serde_json::from_value(json_with_null_denom_units).unwrap();
+        assert_eq!(
+            metadata_with_null_denom_units.denom_units,
+            Vec::<DenomUnit>::default()
+        );
+        assert!(metadata_with_null_denom_units.denom_units.is_empty());
+
+        // Test normal case with provided denom_units
+        let json_with_units = json!({
+            "description": "Test Token",
+            "denom_units": [
+                {
+                    "denom": "utest",
+                    "exponent": 6,
+                    "aliases": ["microtest"]
+                }
+            ],
+            "base": "utest",
+            "display": "TEST",
+            "name": "Test Token",
+            "symbol": "TEST",
+            "uri": "https://test.com",
+            "uri_hash": "hash"
+        });
+
+        let metadata_with_units: DenomMetadata = serde_json::from_value(json_with_units).unwrap();
+        assert_eq!(metadata_with_units.denom_units.len(), 1);
+        assert_eq!(metadata_with_units.denom_units[0].denom, "utest");
+        assert_eq!(metadata_with_units.denom_units[0].aliases.len(), 1);
+        assert_eq!(metadata_with_units.denom_units[0].aliases[0], "microtest");
+
+        // Test with null aliases inside denom_units - should deserialize as empty vec
+        let json_with_null_aliases = json!({
+            "description": "Test Token",
+            "denom_units": [
+                {
+                    "denom": "utest",
+                    "exponent": 6,
+                    "aliases": null
+                }
+            ],
+            "base": "utest",
+            "display": "TEST",
+            "name": "Test Token",
+            "symbol": "TEST",
+            "uri": "https://test.com",
+            "uri_hash": "hash"
+        });
+
+        let metadata_with_null_aliases: DenomMetadata =
+            serde_json::from_value(json_with_null_aliases).unwrap();
+        assert_eq!(metadata_with_null_aliases.denom_units.len(), 1);
+        assert_eq!(
+            metadata_with_null_aliases.denom_units[0].aliases,
+            Vec::<String>::default()
+        );
+        assert!(metadata_with_null_aliases.denom_units[0].aliases.is_empty());
+    }
+
+    #[test]
+    fn query_denom_metadata_with_missing_fields_fails() {
+        // Missing denom_units should be treated as default value (empty vec)
+        let json_missing_denom_units = json!({
+            "description": "Test Token",
+            "base": "utest",
+            "display": "TEST",
+            "name": "Test Token",
+            "symbol": "TEST",
+            "uri": "https://test.com",
+            "uri_hash": "hash"
+        });
+
+        let json_missing_denom_units_metadata: Result<DenomMetadata, Error> =
+            serde_json::from_value(json_missing_denom_units);
+        assert!(json_missing_denom_units_metadata.is_err());
```

Comment:
The comment does not match the assertion. Why does this error?

Reply:
Updated comments